examples/connext_dds/group_coherent_presentation/c++11/group_coherent_subscriber.cxx

```diff
@@ -0,0 +1,205 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2021.  All rights reserved.
+* RTI grants Licensee a license to use, modify, compile, and create derivative
+* works of the software solely for use with RTI Connext DDS. Licensee may
+* redistribute copies of the software provided that all such copies are subject
+* to this license. The software is provided "as is", with no warranty of any
+* type, including any warranty for fitness for any purpose. RTI is under no
+* obligation to maintain or support the software. RTI shall not be liable for
+* any incidental or consequential damages arising out of the use or inability
+* to use the software.
+*/
+
+#include <algorithm>
+#include <iostream>
+
+#include <dds/sub/ddssub.hpp>
+#include <dds/core/ddscore.hpp>
+#include <rti/config/Logger.hpp>
+
+#include "group_coherent.hpp"
+#include "application.hpp"  
+
+using namespace dds::core;
+using namespace dds::core::policy;
+using namespace dds::sub;
+using namespace dds::sub::qos;
+
+template<typename T>
+class CheckpointDRListener : public dds::sub::NoOpDataReaderListener<T>
+{
+public:
+    void on_sample_lost(
+            dds::sub::DataReader<T>&,
+            const dds::core::status::SampleLostStatus& status)
+    {
+        if (status.extensions().last_reason() 
+                == rti::core::status::SampleLostState::lost_by_incomplete_coherent_set()) {
+            std::cout << "Lost " 
+                      << status.total_count_change() 
+                      << " samples in an incomplete coherent set.";
+        }
+    }
+};
+
+void print_coherent_set_info(dds::sub::SampleInfo info)
```

Comment:
```suggestion
void print_coherent_set_info(const dds::sub::SampleInfo& info)
```

This is not only saving an unnecessary copy, it's actually incorrect now because we don't copy the coherent_set_info pointer (this is a 6.1.0 limitation).